ApeWorX/ape
# fix: recursion error when accessing chain ID for non-connected provider

### What I did

Issue causing network ID recursion and double definiton of chain ID / network ID

### How I did it

Call into `super()` when not  connected to get chain ID set from init (the static one) when not  connected

### How to verify it

Make sure you can check the chain ID for a non-connected provider
**NOTE**: Does not really work for non-local networks

### Checklist
